lib/coinbase/staking_operation.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module Coinbase
+  # A representation of a staking operation (stake, unstake, claim rewards, etc). It
+  # may have multiple steps with some being transactions to sign, and others to wait.
+  # @attr_reader [Array<Coinbase::Transaction>] transactions The list of current
+  #   transactions associated with the operation
+  class StakingOperation
+    attr_reader :transactions
+
+    # Returns a new StakingOperation object.
+    # @param model [Coinbase::Client::StakingOperation] The underlying StakingOperation object
+    def initialize(model)
+      @model = model
+
+      @transactions = [model.transaction].map do |transaction_model|
```

Comment:
I assume this is a placeholder because we will eventually support multiple transactions?

Can we add a unit test for `#transactions` and `#sign`?

Reply:
Exactly, let me do a type check & switch so we dont have a breaking change!

And yes!